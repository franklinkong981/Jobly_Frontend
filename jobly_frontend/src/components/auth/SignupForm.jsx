import React, {useState} from "react";
import {useFormik} from "formik";

function SignupForm({signUpFunc}) {
  const formik = useFormik({
    initialValues: {
      username: '',
      password: '',
      firstName: '',
      lastName: '',
      email: ''
    },
    validateOnChange: false,
    validateOnBlur: false,
    async onSubmit(values) {
      await signUpFunc(values);
    }
  });


  return (
    <section className="col-md-4">
      <h1>Sign Up For Jobly</h1>
      <form className="SignUpForm" onSubmit={formik.handleSubmit}>
        <label htmlFor="username=field">Username: </label>
        <input id="username-field" className="SignupForm-username-field" type="text" name="username"
        size="50" value={formik.values.username} onChange={formik.handleChange}/><br/>

        <label htmlFor="password=field">Password: </label>
        <input id="password-field" className="SignupForm-password-field" type="password" name="password"
        size="50" value={formik.values.password} onChange={formik.handleChange}/><br/>

        <label htmlFor="firstName-field">First Name: </label>
        <input id="firstName-field" className="SignupForm-firstName-field" type="text" name="firstName"
        size="50" value={formik.values.firstName} onChange={formik.handleChange}/><br/>

        <label htmlFor="lastName-field">Last Name: </label>
        <input id="lastName-field" className="SignupForm-lastName-field" type="text" name="lastName"
        size="50" value={formik.values.lastName} onChange={formik.handleChange}/><br/>

        <label htmlFor="email-field">Email: </label>
        <input id="email-field" className="SignupForm-email-field" type="email" name="email"
        size="50" value={formik.values.email} onChange={formik.handleChange}/><br/>

        <button className="SignUpForm-submit-button" type="submit">Create Account</button>
      </form>
    </section>
  );
}

export default SignupForm;