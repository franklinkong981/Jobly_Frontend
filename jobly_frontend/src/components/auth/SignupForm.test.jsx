import React from "react";
import {render} from"@testing-library/react";
import {MemoryRouter} from "react-router-dom";

import SignupForm from "./SignupForm.jsx";
import { expect, it } from "vitest";

it("renders without crashing", function() {
  render(
    <MemoryRouter>
      <SignupForm />
    </MemoryRouter>
  );
});

it("matches snapshot", function() {
  const {asFragment} = render(
    <MemoryRouter>
      <SignupForm />
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});