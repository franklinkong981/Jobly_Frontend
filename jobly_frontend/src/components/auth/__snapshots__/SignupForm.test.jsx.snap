// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="SignupForm"
  >
    <div
      class="container col-md-6 offset-md-3 col-lg-4 offset-lg-4"
    >
      <h2
        class="mb-3"
      >
        Create a Jobly Account to Get Started
      </h2>
      <div
        class="card"
      >
        <div
          class="card-body"
        >
          <form>
            <div
              class="form-group"
            >
              <label
                for="SignUpForm-username-field"
              >
                Choose a username
              </label>
              <input
                class="form-control"
                id="SignUpForm-username-field"
                name="username"
                required=""
                type="text"
                value=""
              />
            </div>
            <div
              class="form-group"
            >
              <label
                for="SignUpForm-password-field"
              >
                Choose a password
              </label>
              <input
                class="form-control"
                id="SignUpForm-password-field"
                name="password"
                required=""
                type="password"
                value=""
              />
            </div>
            <div
              class="form-group"
            >
              <label
                for="SignUpForm-first-name-field"
              >
                First name
              </label>
              <input
                class="form-control"
                id="SignUpForm-first-name-field"
                name="firstName"
                required=""
                type="text"
                value=""
              />
            </div>
            <div
              class="form-group"
            >
              <label
                for="SignUpForm-last-name-field"
              >
                Last name
              </label>
              <input
                class="form-control"
                id="SignUpForm-last-name-field"
                name="lastName"
                required=""
                type="text"
                value=""
              />
            </div>
            <div
              class="form-group"
            >
              <label>
                Email
              </label>
              <input
                class="form-control"
                id="SignUpForm-email-field"
                name="email"
                required=""
                type="email"
                value=""
              />
            </div>
            <button
              class="btn btn-primary float-right"
              type="submit"
            >
              Create Account
            </button>
          </form>
          <p>
            Already have an account? 
            <a
              href="/login"
            >
              Log In Now!
            </a>
          </p>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
